app/forms/mappings/Formatters.scala

```diff
@@ -123,11 +123,12 @@ trait Formatters {
         finalResult <- resultWithoutCommas match {
                          case s if s.matches(DecimalRegexp) =>
                            Seq(FormError(key, doNotUseDecimalsKey, args)).asLeft[Int]
-                         case s if Try(s.toLong).isFailure => Seq(FormError(key, nonNumericKey, args)).asLeft[Int]
-                         case s if s.toLong < 1 =>
-                           Seq(FormError(key, outOfRangeKey, Seq(1, ONE_BILLION))).asLeft[Int]
-                         case s if s.toLong > ONE_BILLION =>
-                           Seq(FormError(key, outOfRangeKey, Seq(1, ONE_BILLION))).asLeft[Int]
+                         case s if Try(s.toLong).isFailure && !s.matches(WholeNumber) =>
+                           Seq(FormError(key, nonNumericKey, args)).asLeft[Int]
+                         case s if Try(s.toLong).isFailure =>
+                           Seq(FormError(key, outOfRangeKey, Seq(Int.MinValue, Int.MaxValue))).asLeft[Int]
```

Comment:
Do we need the range implementation here?

Reply:
I added it as a worst-case scenario fallback. Just in case. Was not confortable with throwing a number format exception. Ideally, this should not be triggered ever because of constraints.